src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -310,16 +305,34 @@ private void ProcessConfigureCall(BinderInvocationOperation binderOperation)
                     SymbolEqualityComparer.Default.Equals(_typeSymbols.IConfiguration, @params[1].Type))
                 {
                     ITypeSymbol? type = targetMethod.TypeArguments[0].WithNullableAnnotation(NullableAnnotation.None);
-                    if (type is not INamedTypeSymbol namedType ||
-                        namedType.SpecialType == SpecialType.System_Object)
+
+                    if (!ValidateRootConfigType(BinderMethodSpecifier.Configure, type, binderOperation.Location))
                     {
                         return;
                     }
 
-                    AddRootConfigType(methodGroup: BinderMethodSpecifier.Configure, overload: BinderMethodSpecifier.Configure, namedType, binderOperation.Location);
+                    AddRootConfigType(methodGroup: BinderMethodSpecifier.Configure, overload: BinderMethodSpecifier.Configure, type, binderOperation.Location);
                 }
             }
 
+            private bool ValidateRootConfigType(BinderMethodSpecifier overload, ITypeSymbol? type, Location? location)
```

Comment:
Methods called "Validate" typically imply a `void` return type and throwing if inputs aren't valid.

```suggestion
            private bool IsValidRootConfigType(BinderMethodSpecifier overload, ITypeSymbol? type, Location? location)
```